From 291730c7f2bbe0c5130ee99e31c6b9a242d830da Mon Sep 17 00:00:00 2001
From: Wig Cheng <wig.cheng@technexion.com>
Date: Fri, 3 Dec 2021 14:06:09 +0800
Subject: [PATCH 19/20] arm64: dts: imx8mp-evk: add power pin control to csi1
 with ov5640 sensor

Because power pin enabling must be done before ov5640 initialization, so
adapt gpio node to implement.
---
 .../freescale/imx8mp-evk-tevi-ov5640-csi1.dts | 21 ++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-evk-tevi-ov5640-csi1.dts b/arch/arm64/boot/dts/freescale/imx8mp-evk-tevi-ov5640-csi1.dts
index 0f025e9dbfe0..25bb20f939b9 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-evk-tevi-ov5640-csi1.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-evk-tevi-ov5640-csi1.dts
@@ -15,6 +15,16 @@
 
 #include "imx8mp-evk.dts"
 
+&gpio2 {
+	pinctrl-0 = <&pinctrl_gpio2>;
+	csi_pwr {
+		gpio-hog;
+		gpios = <11 GPIO_ACTIVE_HIGH>;
+		line-name = "CSI_POWER";
+		output-high;
+	};
+};
+
 &i2c2 {
 	/delete-node/ ov5640_mipi@3c;
 
@@ -29,7 +39,7 @@ tevi_ov5640_mipi@3c {
 		compatible = "ovti,ov5640";
 		reg = <0x3c>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_csi0_pwn>, <&pinctrl_csi0_rst>;
+		pinctrl-0 = <&pinctrl_sensor0_pwn>, <&pinctrl_csi0_rst>;
 		clocks = <&clk IMX8MP_CLK_IPP_DO_CLKO2>;
 		clock-names = "xclk";
 		assigned-clocks = <&clk IMX8MP_CLK_IPP_DO_CLKO2>;
@@ -76,9 +86,14 @@ &isi_1 {
 	status = "okay";
 };
 
-
 &iomuxc {
-	pinctrl_csi0_pwn: csi0_pwn_grp {
+	pinctrl_gpio2: gpio2_grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SD1_STROBE__GPIO2_IO11	0x19
+		>;
+	};
+
+	pinctrl_sensor0_pwn: sensor0_pwn_grp {
 		fsl,pins = <
 			MX8MP_IOMUXC_GPIO1_IO05__GPIO1_IO05	0x19
 		>;
-- 
2.25.1

